### DISCUSSION TITLE
Installations paths in settings

### DISCUSSION THREAD

**Discussion by AFulgens** [microsoft/winget-cli#4110] [Q&A]
https://github.com/microsoft/winget-cli/blob/master/doc/Settings.md tells me that I can define installations paths for (non-)portable installations with
```json
"installBehavior": {
    "portablePackageUserRoot": "C:/Users/FooBar/Packages",
    "portablePackageMachineRoot": "C:/Program Files/Packages/Portable"
},
```

The documentation also tells me that these must be absolute paths (totally legitimate restriction), with defaulting to `%LOCALAPPDATA%/Microsoft/WinGet/Packages/` and `%PROGRAMFILES%/WinGet/Packages/` respectively, in case they are not defined.

Thus, I conclude that I can use CMD-style variables in the paths, i.e., I would expect `%USERPROFILE%/.winget/` to work.

Alas, it results in a validation failure:
```
PS C:\Users\FooBar> winget settings
Unexpected error while loading settings. Please verify your settings by running the 'settings' command.
The following failures were found validating the settings:
Invalid field value. Field: .installBehavior.defaultInstallRoot Value: %USERPROFILE%/.winget/
Invalid field value. Field: .installBehavior.portablePackageUserRoot Value: %USERPROFILE%/.winget/
```

Either the documentation should say that it must be an absolute path _and_ no variables are allowed, or winget should dereference variables.

Am I missing something here?

**Comment by Gijsreyn** [comment]
@AFulgens I was just about to add it to the documentation when I put on my glasses. It's there:

![image](https://github.com/user-attachments/assets/009716a6-09e2-4b68-a4ee-01a5d4f3deda)

**Comment by AFulgens** [comment]
It is there that it must be an absolute path.

If I have an envrionment variable via

```
SET test="C:/Program Files/Packages/Portable"
```

then `%test%` is an absolute path.

My problem is that `%USERPROFILE%/.winget/` defines by definition an absolute path (`C:/<Users>/<Current Username>/.winget`, whereby `<Current Username>` is resolved, as well as `<Users>` to be the local-correct alias). What the problem is that (environment) variables are not resolved, even though the documentation hints at this with `%PROGRAMFILES%` being in the "if value is not set".

To quote from my original post:

> Either the documentation should say that it must be an absolute path _and_ no variables are allowed, or winget should dereference variables.

**Reply by Gijsreyn** [reply]
I got you wrong; now I understand.
